include/internal/decoder.h

```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright 2023 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#ifndef OSSL_INTERNAL_DECODER_H
+# define OSSL_INTERNAL_DECODER_H
+# pragma once
+
+void *ossl_decoder_cache_new(OSSL_LIB_CTX *ctx);
+void ossl_decoder_cache_free(void *vcache);
+int ossl_decoder_cache_flush(OSSL_LIB_CTX *libctx);
```

Comment:
Why were these lines not placed in include/crypto/decoder.h, *just like everything else that belongs in libcrypto only*?

Reply:
Good question. They probably should have been.

Reply:
Fix in #21733